File: app/src/main/java/com/kongjak/ggcj/Tools/FileAdapter.java
package com.kongjak.ggcj.Tools;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.kongjak.ggcj.R;

import java.util.ArrayList;

public class FileAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    Context context;
    private ArrayList<Files> FileArrayList;

    public FileAdapter(ArrayList<Files> FileArrayList) {
        this.FileArrayList = FileArrayList;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.file_card, parent, false);

        return new MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, final int position) {
        MyViewHolder myViewHolder = (MyViewHolder) holder;

        final Files data = FileArrayList.get(position);

        myViewHolder.title.setText(FileArrayList.get(position).title);

        myViewHolder.mCardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                //Toast.makeText(view.getContext(), data.getUrl(), Toast.LENGTH_SHORT).show();
                view.getContext().startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(data.getUrl())));
            }
        });
    }

    @Override
    public int getItemCount() {
        return FileArrayList.size();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder {
        CardView mCardView;
        TextView title;

        MyViewHolder(View view) {
            super(view);
            title = view.findViewById(R.id.item_dl);
            mCardView = (CardView) view.findViewById(R.id.dl_card);
        }
    }
}

File: app/src/main/java/com/kongjak/ggcj/Tools/FileAdapter.kt
package com.kongjak.ggcj.Tools

import android.content.Intent
import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.kongjak.ggcj.R
import java.util.*

class FileAdapter(private val FileArrayList: ArrayList<Files>) :
    RecyclerView.Adapter<FileAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
        ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.file_item, parent, false))

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val data = FileArrayList[position]
        holder.title.text = data.title
        holder.mCardView.setOnClickListener {
            it.context.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(data.url)))
        }
    }


    override fun getItemCount(): Int {
        return FileArrayList.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val mCardView = itemView.findViewById<CardView>(R.id.dl_card)
        val title = itemView.findViewById<TextView>(R.id.item_dl)
    }
}